src/main/java/com/es/phoneshop/web/AddCartItemFromSearchPage.java

```diff
@@ -0,0 +1,83 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.DefaultCartService;
+import com.es.phoneshop.model.cart.OutOfStockException;
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.model.product.ProductDao;
+import com.es.phoneshop.model.product.feature.RecentlyViewedProductsService;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.List;
+
+
+public class AddCartItemFromSearchPage extends HttpServlet {
```

Comment:
Bad name, at least it should contain Servlet, please correct it. Btw why not to use doPost of plp servlet for this functionality.